### ISSUE TITLE
NameError: uninitialized constant PgHero::Connection

### ISSUE THREAD

**Issue by zeha** [ankane/pghero#233]
Hi,

we've been running the Debian package `2.2.0-1536002635.42c92b9e.stretch` successfully for a while, but after upgrading to `2.2.0-1537533410.29613939.stretch` the rake cronjobs fail with this error:

```
Capturing query stats for dwh...
NameError: uninitialized constant PgHero::Connection

rake aborted!
NameError: uninitialized constant PgHero::Connection
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero/database.rb:72:in `connection_model'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero/methods/basic.rb:80:in `connection'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero/methods/basic.rb:36:in `select_all'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero/methods/basic.rb:68:in `select_one'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero/methods/basic.rb:18:in `database_name'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero/methods/query_stats.rb:95:in `capture_query_stats'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero.rb:122:in `block in capture_query_stats'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero.rb:158:in `block in each_database'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero.rb:156:in `each'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero.rb:156:in `each_database'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/pghero.rb:119:in `capture_query_stats'
/opt/pghero/vendor/bundle/ruby/2.5.0/bundler/gems/pghero-80960454505d/lib/tasks/pghero.rake:4:in `block (2 levels) in <top (required)>'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/cli/exec.rb:74:in `load'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/cli/exec.rb:74:in `kernel_load'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/cli/exec.rb:27:in `run'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/cli.rb:360:in `exec'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/vendor/thor/lib/thor.rb:369:in `dispatch'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/cli.rb:20:in `dispatch'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/vendor/thor/lib/thor/base.rb:444:in `start'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/cli.rb:10:in `start'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/exe/bundle:30:in `block in <top (required)>'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
/opt/pghero/vendor/bundle/ruby/2.5.0/gems/bundler-1.15.2/exe/bundle:22:in `<top (required)>'
/opt/pghero/vendor/bundle/ruby/2.5.0/bin/bundle:22:in `load'
/opt/pghero/vendor/bundle/ruby/2.5.0/bin/bundle:22:in `<main>'
Tasks: TOP => pghero:capture_query_stats
(See full trace by running task with --trace)
```

As far as I can tell the older version does not have commit bc8bc03757769cc8631735b5367387c19c22d684, but I'm not sure this is the cause. 

Thanks!

**Comment by ankane** [comment]
Hey @zeha, definitely could be related, but I'm unable to reproduce with a fresh install on Vagrant: https://github.com/pghero/pghero/commit/eb8b8d635897a47f775d77517593405d94b2d691#diff-23b6f443c01ea2efcb4f36eedfea9089R34

**Comment by Gurpartap** [comment]
Getting the exact same error and trace. Maybe the only difference from the vagrant Debian stretch commands I have is that I've defined PGHERO_STATS_DATABASE_URL and [Multiple Databases](https://github.com/ankane/pghero/blob/master/guides/Linux.md#multiple-databases) (using `cat pghero.yml | pghero run sh -c "cat > config/pghero.yml"` command.

```
root@linode123 # cat /proc/version
Linux version 4.18.8-x86_64-linode117 (maker@build.linode.com) (gcc version 4.9.2 (Debian 4.9.2-10+deb8u1)) #1 SMP PREEMPT Tue Sep 18 18:48:25 UTC 2018
```

```
postgres@linode123:/apps/pghero $ psql -c 'SELECT version();'
                                                             version                                                              
----------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 11.0 (Debian 11.0-1.pgdg90+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit
(1 row)
```

```
root@linode123 # apt search pghero
Sorting... Done
Full Text Search... Done
pghero/unknown,now 2.2.0-1537533410.29613939.stretch amd64 [installed]
  A performance dashboard for Postgres
```

```
root@linode123 # pghero config
APP_GROUP=pghero
APP_HOME=/opt/pghero
APP_NAME=pghero
APP_RUNNER_CLI=systemctl
APP_RUNNER_TYPE=systemd
APP_SAFE_NAME=pghero
APP_USER=pghero
APP_WIZARDS=
DATABASE_URL=postgres://postgres:<password>@localhost:5432/pghero?sslmode=disable
GEM_PATH=/opt/pghero/vendor/bundle/ruby/2.5.0:
HOME=/home/pghero
LANG=en_US.UTF-8
OLDPWD=/
ORIGINAL_PWD=/apps/pghero
PATH=/opt/pghero/bin:/opt/pghero/vendor/bundle/bin:/opt/pghero/vendor/bundle/ruby/2.5.0/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:.:/usr/sbin
PGHERO_PASSWORD=<pgheropassword>
PGHERO_STATS_DATABASE_URL=postgres://postgres:<password>@localhost:5432/pghero?sslmode=disable
PGHERO_USERNAME=admin
PORT=3001
PWD=/opt/pghero
RACK_ENV=production
RAILS_ENV=production
RAILS_LOG_TO_STDOUT=disabled
RAILS_SERVE_STATIC_FILES=enabled
ROOT_PATH=/
SECRET_KEY_BASE=<secret-key-base>
SHLVL=0
```

**Comment by ankane** [comment]
@zeha @Gurpartap Still can't repro, but just pushed an attempt to fix it. Should be available in the next 30 min or so. Please let me know if that solves it.

**Comment by Gurpartap** [comment]
@ankane It was bad yaml syntax in the multiple dbs file, `pghero.yml`.

No error after fixing the syntax.

Perhaps a graceful error detail would be helpful for others.

**Comment by ankane** [comment]
Cleaning up issues

**Comment by zeha** [comment]
I've again checked your fix (4b3b2d08c152d2e67bbce1b704a1efe51321d81b) in docker, and can confirm it works there. Could I ask you to make a release with that in?
Thanks!
